.github/workflows/docker.yml

```diff
@@ -0,0 +1,40 @@
+name: Build & Push (main)
```

Comment:
should we go with a slightly more descriptive name? Build & Push simapp image?

Reply:
Totally open to suggestions, maybe - "Docker build & push simapp/simd"

I'm thinking it may be useful to also add a [path filter](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) on go src files. E.g. adding the below to the `on: push` event trigger

```
paths:
  - '**.go'
```

Reply:
Yeah, that definitely makes sense to me.